src/mod/server.mod/isupport.c

```diff
@@ -0,0 +1,451 @@
+    result += digit;
+  }
+  /* disallow \x00 */
+  return result ? result : -1;
+}
+
+static const char *isupport_encode(const char *value) {
+  static char buf[512];
+  int i, j;
+
+  for (i = j = 0; i < strlen(value) && j < sizeof buf - strlen("\\xHH") - 1; i++) {
+    if ((unsigned char)value[i] < 33 || (unsigned char)value[i] > 126)
+      j += sprintf(buf + j, "\\x%02hhx", (unsigned char)value[i]);
+    else
+      buf[j++] = value[i];
+  }
+  buf[j] = '\0';
+  return buf;
+}
+
+static size_t isupport_decode(char *buf, size_t bufsize,
+    const char *value, size_t len)
+{
+  const char *ch;
+  char *bufptr = buf;
+  int state = 0, hexchr;
+
+  for (ch = value; (ch - value) < len && (bufptr - buf) < bufsize - 1; ch++) {
+    if (state == 0 && *ch == '\\') {
+      state++;
+    } else if (state == 1) {
+      if (*ch != 'x')
+        goto rollback;
+      state++;
+    } else if (state == 2) {
+      hexchr = isupport_hex2chr(ch, 2);
+      if (hexchr == -1)
+        goto rollback;
+      *bufptr++ = hexchr;
+      ch++;
+      state = 0;
+    } else {
+rollback:
+      if (state > 0)
+        *bufptr++ = '\\';
+      if (state > 1)
+        *bufptr++ = 'x';
+      state = 0;
+      *bufptr++ = *ch;
+    }
+  }
+  *bufptr = '\0';
+  return (size_t)(bufptr - buf);
+}
+
+static void isupport_parse(const char *str,
+    void (*set_cb)(const char *key, size_t keylen, const char *v, size_t vlen))
+{
+  static char buf[512];
+  size_t keylen, valuelen;
+  const char *key, *value;
+
+  while (1) {
+    while (*str == ' ')
+      str++;
+    /* :are supported by this server */
+    if (!*str || *str == ':')
+      break;
+    key = str;
+    keylen = strcspn(key, "= ");
+    if (key[0] == '-') {
+      putlog(LOG_MISC, "*", "isupport unsetting with -KEY is not supported, please report this");
+      str = key + keylen;
+    } else {
+      value = key + keylen + (key[keylen] == '=');
+      valuelen = strcspn(value, " ");
+      valuelen = isupport_decode(buf, sizeof buf, value, valuelen);
+      if (set_cb) {
+        set_cb(key, keylen, buf, valuelen);
+      }
+      str = value + valuelen;
+    }
+  }
+}
+
+
+/*** Module utility functions ***/
+
+
+void isupport_clear(void) {
+  struct isupport *data = isupport_list, *next;
+
+  isupport_list = NULL;
+
+  while (data) {
+    next = data->next;
+    isupport_free(data);
+    data = next;
+  }
+}
+
+void isupport_clear_values(int cleardefaultvalues) {
+  struct isupport *next;
+
+  for (struct isupport *data = isupport_list; (next = data ? data->next : NULL, data); data = next) {
+    if ((cleardefaultvalues && data->defaultvalue) || (!cleardefaultvalues && data->value)) {
+      if (cleardefaultvalues && data->value) {
+        nfree(data->defaultvalue);
+      } else if (!cleardefaultvalues && data->defaultvalue) {
+        nfree(data->value);
+      } else {
+        /* entry will be empty, delete it */
+        del_record(data);
+      }
+    }
+  }
+}
+
+void isupport_init(void) {
```

Comment:
I will use the following default values:
```
* PREFIX=(ov)@+
* CHANTYPES=#& because + and = is non-standard and not widely used
* CHANMODES=b,k,l,imnpst
* MODES=3
* MAXCHANNELS=10
* CASEMAPPING=rfc1459
* CHANNELLEN=80 because that is eggdrop's internal limit now (freenode has 50)
* TOPICLEN=250
* NICKLEN=32 because we get to know this value too late to truncate eggdrop's nick before connecting anyway
* STATUSMSG=@+
* KICKLEN=250 as a reasonable default (== TOPICLEN)
```
This list includes the most commonly used values (most important) and is based on my experience, most popular irc networks, and the excellent https://stats.ircdocs.horse/isupport/ by @danieloaks